specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensions.json

```diff
@@ -0,0 +1,723 @@
+          "x-ms-client-flatten": true,
+          "description": "Properties of an Extension resource",
+          "properties": {
+            "extensionType": {
+              "description": "Type of the Extension, of which this resource is an instance of.  It must be one of the Extension Types registered with Microsoft.KubernetesConfiguration by the Extension publisher.",
+              "type": "string"
+            },
+            "autoUpgradeMinorVersion": {
+              "description": "Flag to note if this extension participates in auto upgrade of minor version, or not.",
+              "type": "boolean",
+              "default": true
+            },
+            "releaseTrain": {
+              "description": "ReleaseTrain this extension participates in for auto-upgrade (e.g. Stable, Preview, etc.) - only if autoUpgradeMinorVersion is 'true'.",
+              "type": "string",
+              "default": "Stable"
+            },
+            "version": {
+              "description": "Version of the extension for this extension, if it is 'pinned' to a specific version. autoUpgradeMinorVersion must be 'false'.",
+              "type": "string"
+            },
+            "scope": {
+              "description": "Scope at which the extension is installed.",
+              "$ref": "#/definitions/Scope"
+            },
+            "configurationSettings": {
+              "description": "Configuration settings, as name-value pairs for configuring this extension.",
+              "type": "object",
+              "x-nullable": true,
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            "configurationProtectedSettings": {
+              "description": "Configuration settings that are sensitive, as name-value pairs for configuring this extension.",
+              "type": "object",
+              "x-nullable": true,
+              "x-ms-secret": true,
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            "provisioningState": {
+              "description": "Status of installation of this extension.",
+              "type": "string",
+              "readOnly": true,
+              "$ref": "#/definitions/provisioningState"
+            },
+            "statuses": {
+              "description": "Status from this extension.",
+              "type": "array",
+              "x-nullable": true,
+              "items": {
+                "$ref": "#/definitions/ExtensionStatus"
+              }
+            },
+            "errorInfo": {
+              "description": "Error information from the Agent - e.g. errors during installation.",
+              "type": "object",
+              "readOnly": true,
+              "x-nullable": true,
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorDetail"
+            },
+            "identityLocation": {
```

Comment:
Is this how we capture the identity location at RP? I thought this was outside the property bag

Reply:
Identity property is needed by Azure and is outside the properties, which is the requirement.  However, IdentityLocation is additional information that we need when we request Identity creation - to specify its location.  Since this is for our consumption, it goes inside properties.